fix: remove return of token usage for the correcting agent for BioChatter light

fcarli opened the pull request:
Erroneously, https://github.com/biocypher/biochatter/pull/304 introduced a return of the token usage for the correcting agent, breaking compatibility with the BioChatter-light interface. This PR addresses the issue by reverting to the original behavior.

slobentanzer commented:
@fcarli why erroneously? I thought you wanted to change / simplify that 😅

The correction functionality is outdated anyways, we should remove that soon and not spend too much time on maintaining it (IMO). Should be much more generic in an agentic setting. Happy to either roll back or change it in biochatter light. You can make that decision.

Minor comment: make sure to sync with the current version of `main` before branching, this PR is outdated.

fcarli commented:
> @fcarli why erroneously? I thought you wanted to change / simplify that 😅
> 
> The correction functionality is outdated anyways, we should remove that soon and not spend too much time on maintaining it (IMO). Should be much more generic in an agentic setting. Happy to either roll back or change it in biochatter light. You can make that decision.
> 
> Minor comment: make sure to sync with the current version of `main` before branching, this PR is outdated.

The previous PR simplified token counting across providers for the primary query (that is still there). I had also added token counting for the correcting agent, which wasn’t previously tracked. I’m now reverting just that part, as it’s the lowest-effort change and I agree with you that the current functionality is outdated.

I think that we should, at some point, address the return issue as I introduced in https://github.com/biocypher/biochatter/pull/297 because it will probably make the code more robust to these types of issues.
